File: day09.java
//921 - Minimum Add to make parenthesis valid
class Solution {
    public int minAddToMakeValid(String s) {
        int open=0,close=0;
        for(int i =0;i<s.length();i++){
            //count the opening
            if(s.charAt(i)=='('){
                open++;
                //if it has a balanced counterhalf then pop
            }else if(open>0 && s.charAt(i)==')') {
                open--;
            }
                //if it is standalone closing then count
            else{
                close++;
            }
        }
        //return tthe leftover sum
        return open+close;
    }
}

File: day09.go
//921 - minimum add to make parenthesis valid
//My Go programming approach
func minAddToMakeValid(s string) int {
    close:=0
    open:=0
    for _, char := range s {
        if char == '(' {
            open++
        } else if open > 0 && char == ')' {
            open--
        } else {
            close++
        }
    }
    return open + close
}
